### PR TITLE
Update continue reading and main page cells

### PR DESCRIPTION
**Pull request by coreyfloyd**
https://phabricator.wikimedia.org/T125465

Also some boyscouts - see commit comments

### CODE REVIEW

File: Wikipedia/Code/WMFExploreSectionSchema.m [hunk 1]
```diff
@@ -425,6 +443,15 @@ - (nullable WMFExploreSection*)continueReadingSection {
     return nil;
 }
 
+- (nullable)existingCOntinueReadingSection {
```
Comment (montehurd) [inline Wikipedia/Code/WMFExploreSectionSchema.m]: Typo.

**Comment by bgerstle** [comment]
was just testing this w/ tweaks on, but "Continue Reading" seems to not display an accurate timestamp. here, it shows 85 minutes even though i just read it:

- Disable "Open last article"
- Enable "always show continue reading"
- Open "Gerald Ford"
- Quit the app
- Relaunch:
- :x: "Gerald Ford" automatically opens on launch :confused: 
- :x: "Continue Reading" shows 85 minutes

image


works, but the main page just looks... weird

image

**Comment by coreyfloyd** [comment]
> :x: "Continue Reading" shows 85 minutes

@bgerstle The date is based on the last time you put the app resigned active - as opposed to the last time you force quite the simulator.

**Comment by bgerstle** [comment]
@coreyfloyd ahh lemme try that

**Comment by coreyfloyd** [comment]
> :x: "Gerald Ford" automatically opens on launch :confused:

@bgerstle if less than 24 hours, you are supposed to be pushed to the last article you read

**Comment by bgerstle** [comment]
even if "open last article" is off?

**Comment by coreyfloyd** [comment]
> even if "open last article" is off?

@bgerstle no not in that case - if that is what is happening let me know

**Comment by bgerstle** [comment]
@coreyfloyd just confirmed: it's opening the last article even if that tweak is off and that Continue Reading timestamp is accurate if I put app in the background first.

:+1: will file Last Open Article thing as a separate issue (since it's only a problem w/ the tweak, not the feature)

**Comment by montehurd** [comment]
@bgerstle 
> also the main page just looks... weird. I wish there was more we could do

There is! All we have to do is a fetch for that language's "Main page" title with WMFArticlePreviewFetcher, then format the result using a search results style cell. On enwiki for today the result looks like this:

screen shot 2016-02-10 at 11 27 31 am

**Comment by bgerstle** [comment]
Continue reading screencap for posterity: 

image

**Comment by bgerstle** [comment]
@montehurd true, i guess that's out of scope for this PR though. "Main page of a Wikimedia project" is the wikidata description I suppose?

**Comment by montehurd** [comment]
@bgerstle 

> "Main page of a Wikimedia project" is the wikidata description I suppose?

Correct. But we get a nice image too :)

**Comment by bgerstle** [comment]
@montehurd this is good IMO, can you review/merge?

File: Wikipedia/Code/WMFContinueReadingSectionController.m [hunk 2]
```diff
@@ -1,6 +1,6 @@
 
 #import "WMFContinueReadingSectionController.h"
-#import "WMFContinueReadingTableViewCell.h"
+#import "WMFArticleListTableViewCell.h"
```
Comment (montehurd) [inline Wikipedia/Code/WMFContinueReadingSectionController.m]: Can we delete WMFContinueReadingTableViewCell now?

Comment (bgerstle) [inline Wikipedia/Code/WMFContinueReadingSectionController.m]: and main page!

**Comment by coreyfloyd** [comment]
@bgerstle @montehurd cells removed

**Comment by montehurd** [comment]
:-1: 

Looking good, but just a couple small issues:

-----

The Wikidata description's first letter isn't being capitalized:
screen shot 2016-02-10 at 6 11 54 pm

-----

And Main page isn't quite doing what the ticket describes: 
> Do a search-type query for the exact "continue reading" page title and the localized "Main page" title. (presently using for Disambiguation titles)

The WMFDisambiguationPagesViewController does what amounts to a search query but one that wants info about an exact list of titles. 

     [[WMFArticlePreviewDataSource alloc] initWithTitles:self.article.disambiguationTitles
        site:self.article.site
        dataStore:dataStore
        fetcher:[[WMFArticlePreviewFetcher alloc] init]];

In this case we just need to pass it one title - the language wiki's "main page" title.

Then, per the phab ticket, we'll see this:
screen shot 2016-02-10 at 6 10 20 pm

Instead of:
screen shot 2016-02-10 at 6 19 13 pm

**Comment by coreyfloyd** [comment]
@montehurd done

**Comment by montehurd** [comment]
It's beautiful! Tested in Hebrew - looking good! :+1:

**Pull request merged**